cxx/tests/test_hirm_animals.cc

```diff
@@ -29,10 +29,10 @@ int main(int argc, char **argv) {
   printf("--- initialized HIRM --- \n");
   incorporate_observations(hirm, encoding_unary, observations_unary);
   printf("--- incorporated observations --- \n");
-  int n_obs_unary = 0;
+  size_t n_obs_unary = 0;
```

Comment:
Either using size_t or std::size() for computing sizes works. I'm flexible either way, but perhaps we want to move everything to size_t?

Reply:
I'd lean towards `size_t` (as discussed offline). I'll send a follow-up.